File: split/makevcenter.pas
procedure makevcenter(q: halfword);
var
  v: halfword;
  delta: scaled;
begin
  v := mem[q + 1].hh.lh;
  if mem[v].hh.b0 <> 1 then
    confusion(539);
  delta := mem[v + 3].int + mem[v + 2].int;
  mem[v + 3].int := fontinfo[22 + parambase[eqtb[3937 + cursize].hh.rh]].int + half(delta);
  mem[v + 2].int := delta - mem[v + 3].int;
end;

File: cpp/makevcenter.cpp
#include "makevcenter.h"
#include "confusion.h"	
#include "half.h"	

void makevcenter(halfword q)
{
	halfword v = info(nucleus(q));
	if (type(v) != vlist_node)
		confusion(539); //vcenter
	scaled delta = height(v)+depth(v);
	height(v) = axis_height(cursize)+half(delta);
	depth(v) = delta-height(v);
}

